transform.ml

```diff
@@ -0,0 +1,89 @@
+open Instr
+
+let remove_empty_jmp prog =
+  let rec remove_empty_jmp pc =
+    let pc' = pc + 1 in
+    if pc' = Array.length prog then [prog.(pc)] else
+      match (prog.(pc), prog.(pc')) with
+      | (Goto l1, Label l2) when l1 = l2 ->
+          remove_empty_jmp (pc+2)
+      | (_, _) ->
+          prog.(pc) :: remove_empty_jmp (pc')
+  in
+  let cleaned = remove_empty_jmp 0 in
+  Array.of_list cleaned
+
+let remove_dead_code prog entry=
+  let dead_code =
+    let init_state = ((), entry) in
+    let merge _ _ = None in
+    let update _ _ = () in
+    Analysis.forward_analysis init_state merge update prog
+  in
+  let rec remove_dead_code pc =
+    if pc = Array.length prog then [] else
+      match dead_code.(pc) with
+      | None -> remove_dead_code (pc+1)
+      | Some _ -> prog.(pc) :: remove_dead_code (pc+1)
+  in
+  Array.of_list (remove_dead_code 0)
+
+(* TODO: allow pruning of the true branch, but we need not expression for that *)
```

Comment:
I actually already added the != in #10 and also thought about the expression rewriting. but it seems messy, since we'll have to support every expression which evaluates to bool and it will not be clear where the code comes from. Maybe we should have invalidate_on_true and invaliate_on_false?